src/modules/Core/Wallets/action.js

```diff
@@ -29,15 +30,35 @@ export const updateWalletsRequest = () => (dispatch: Dispatch, getState: GetStat
   const account = CORE_SELECTORS.getAccount(state)
   const { activeWalletIds, archivedWalletIds, currencyWallets } = account
 
-  for (const walletId: string of Object.keys(currencyWallets)) {
-    const edgeWallet: EdgeCurrencyWallet = currencyWallets[walletId]
-    if (edgeWallet.type === 'wallet:ethereum') {
-      if (state.ui.wallets && state.ui.wallets.byId && state.ui.wallets.byId[walletId]) {
-        const enabledTokens = state.ui.wallets.byId[walletId].enabledTokens
-        edgeWallet.enableTokens(enabledTokens)
+  getReceiveAddresses(currencyWallets).then((receiveAddresses) => {
```

Comment:
Why do you need to update the `receiveAddresses` for all wallets and not just the affected wallet?

Reply:
"Updating" a wallet includes getting an `EdgeReceiveAddress`
When updating all the wallets, all the wallets need `EdgeReceiveAddress`es
It's more of a requirement due to how I'm "updating" wallets than anything else